addons/hr_expense/models/hr_expense.py

```diff
@@ -321,21 +372,12 @@ def create_expense_from_attachments(self, attachment_ids=None, view_type='tree')
             })
             attachment.register_as_main_attachment()
             expenses += expense
-        if len(expenses) == 1:
-            return {
-                'name': _('Generated Expense'),
-                'view_mode': 'form',
-                'res_model': 'hr.expense',
-                'type': 'ir.actions.act_window',
-                'views': [[False, 'form']],
-                'res_id': expenses[0].id,
-            }
         return {
-            'name': _('Generated Expenses'),
-            'domain': [('id', 'in', expenses.ids)],
+            'name': _('Generate Expenses'),
             'res_model': 'hr.expense',
             'type': 'ir.actions.act_window',
             'views': [[False, view_type], [False, "form"]],
+            'context': {'search_default_my_expenses': 1, 'search_default_no_report': 1},
         }
```

Comment:
We need to have a uniform way of dealing with this kind of things in Odoo, I'm not sure all PO agree on how this should behave

Reply:
I see your point.. maybe it's inconsistent if this is done differently somewhere else.. but here it makes sense to me. One is uploading expenses from the list view, so makes sense they see only their own expenses. But if we remove this, it won't break anything as well. For now I will leave it and we'll see it the future. Thanks for noticing and feedback :)